### PR TITLE
Add intro tutorial which pulls content from interactive examples

### PR DESCRIPTION
**Pull request by corranwebster**
This extracts and links to the introduction to traits tutorial in traits/examples and makes it available via the documentation.

Done this way rather than copying so we have one source for the actual content.

No new, code, just documentation.

**Checklist**
- [x] ~Tests~
- [x] ~Update API reference (`docs/source/traits_api_reference`)~
- [x] ~Update User manual (`docs/source/traits_user_manual`)~
- [x] ~Update type annotation hints in `traits-stubs`~

### CODE REVIEW

File: docs/source/traits_tutorial/introduction.rst
Comment (mdickinson) [inline docs/source/traits_tutorial/introduction.rst:5]:
```suggestion
This tutorial is intended to introduce you to the basics of Traits
```

File: docs/source/traits_tutorial/introduction.rst
Comment (mdickinson) [inline docs/source/traits_tutorial/introduction.rst:6]:
```suggestion
and to give you an idea of what capabilities the Traits library provides.
```

File: docs/source/traits_tutorial/introduction.rst
Comment (mdickinson) [inline docs/source/traits_tutorial/introduction.rst:9]:
```suggestion
the Python scientific ecosystem, such as NumPy.
```

**Review by mdickinson — COMMENTED** [review]
A few typos noted in passing; I haven't tested the changes in either the built documentation or the demo app.

**Comment by mdickinson** [comment]
Closing and re-opening to retrigger CI build

File: docs/source/index.rst
Comment (kitchoi) [inline docs/source/index.rst:10]: This being at the top makes it more likely the first thing a beginner would read. But I think the Introduction section is more suitable as the first thing to read because it provides a higher-level summary of the capabilities of Traits: https://docs.enthought.com/traits/traits_user_manual/intro.html#introduction

I'd think this tutorial needs to be read _after_ the introduction because it assumes one knows what Traits is for, is interested in it, and wanted to start learning to use it.

Once we establish a structure of our documentation to contain gentle guides/tutorials/API references/example snippets, it will likely get replicated to other ETS repository. So it might be worth thinking about this earlier. 

(I personally like sqlalchemy's documentation (e.g. [Version 1.3](https://docs.sqlalchemy.org/en/13/)). Its landing page guides you through an overview, installation, and then provide links to tutorial and API references.)

**Review by kitchoi — COMMENTED** [review]
I have tested this with the demo application. Looks fine in general but the code snippet looks slightly odd but I think it is an existing issue, hence orthogonal:
Screenshot 2021-01-04 at 10 39 01

I have also tested the HTML sphinx documentation locally. Links are working and the rendering looks fine there.

However I find the "Tutorials" link is mentioned too early in the main page compared to the high-level introduction (see another comment):
Screenshot 2021-01-04 at 11 19 55

**Comment by mdickinson** [comment]
I merged master in to fix (I hope) the unrelated CI failures.

**Comment by corranwebster** [comment]
> I have tested this with the demo application. Looks fine in general but the code snippet looks slightly odd but I think it is an existing issue, hence orthogonal:

That looks like it's not picking up the CSS for the HTML view?

**Comment by mdickinson** [comment]
I'd like to get this in for the impending Traits 6.2 release. There are some important discussions going on, and I don't want to shut those down. But on the basis that

- The docs _with_ the tutorial are a (huge) net improvement over the docs _without_ the tutorial, and
- We can continue to refine the documentation after the release

I'd like to merge this. @kit, @corranwebster are there any points that you think it's really important to address before this is merged?

**Comment by kitchoi** [comment]
Agreed that a doc with tutorial is a net improvement. But the order in the index feels odd enough to me that I'd propose somehow making tutorial follow the "Introduction". I will have a look in how the TOC can be changed to make that happen.

**Comment by kitchoi** [comment]
See #1394

**Comment by kitchoi** [comment]
@mdickinson Please do go ahead to close/merge #1394 and this one as you see fit. When I opened #1394, I really did not think it would be objected to because the ordering was apparently odd to me - it was not my intention to slow down progress.

**Comment by mdickinson** [comment]
I took the liberty of fixing a hyphenation nitpick; will merge when CI signs off.

I'll open a new issue for refactoring of the landing page.

**Comment by mdickinson** [comment]
> I'll open a new issue for refactoring of the landing page

Opened #1413.

**Review by mdickinson — APPROVED** [review]

**Pull request merged**
